detail.py

```diff
@@ -0,0 +1,11 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from catalog import models as cmod
+
+@view_function
+def process_request(request, p:cmod.Product = None):
+
```

Comment:
maybe add a redirect in case the product is None
if product is None:
        return HttpResponseRedirect('/catalog/index/')

Reply:
Bingo. That's very helpful; thank you.